src/lib/storage/constraints/single_constraint_checker.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <algorithm>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "operators/validate.hpp"
+#include "storage/constraints/row_templated_constraint_checker.hpp"
+#include "storage/segment_accessor.hpp"
+
+namespace opossum {
+
+template <typename T>
+class SingleConstraintChecker : public RowTemplatedConstraintChecker<T> {
+ public:
+  SingleConstraintChecker(const Table& table, const TableConstraintDefinition& constraint)
+      : RowTemplatedConstraintChecker<T>(table, constraint) {
+    Assert(constraint.columns.size() == 1, "Only one column constraints allowed for SingleConstraintChecker");
+  }
+
+  virtual std::shared_ptr<std::vector<T>> get_inserted_rows(std::shared_ptr<const Table> table_to_insert) const {
+    auto values = std::make_shared<std::vector<T>>();
+
+    for (const auto& chunk : table_to_insert->chunks()) {
+      const auto segment = chunk->segments()[this->_constraint.columns[0]];
```

Comment:
Unfortunately the compiler forces us to do because that member lies in the templated parent class.

```[ 14%] Building CXX object src/lib/CMakeFiles/hyrise.dir/storage/constraints/unique_checker.cpp.o
In file included from /home/moritz/dev/hpi/hyrise/src/lib/storage/constraints/unique_checker.cpp:8:
/home/moritz/dev/hpi/hyrise/src/lib/storage/constraints/single_constraint_checker.hpp: In member function ‘virtual std::shared_ptr<std::vector<T> > opossum::SingleConstraintChecker<T>::get_inserted_rows(std::shared_ptr<const opossum::Table>) const’:
/home/moritz/dev/hpi/hyrise/src/lib/storage/constraints/single_constraint_checker.hpp:26:46: error: ‘_constraint’ was not declared in this scope
       const auto segment = chunk->segments()[_constraint.columns[0]];
                                              ^~~~~~~~~~~
/home/moritz/dev/hpi/hyrise/src/lib/storage/constraints/single_constraint_checker.hpp:26:46: note: suggested alternative: ‘to_string’
       const auto segment = chunk->segments()[_constraint.columns[0]];
                                              ^~~~~~~~~~~
                                              to_string```